crates/wit-component/tests/interfaces/variants/variants.wat

```diff
@@ -18,46 +18,61 @@
   (type (;16;) (option (type 0)))
   (type (;17;) (option float32))
   (type (;18;) (option (type 3)))
-  (type (;19;) (option (type 12)))
-  (type (;20;) (func (param "a" (type 12)) (param "b" (type 14)) (param "c" (type 15)) (param "d" (type 16)) (param "e" (type 17)) (param "f" (type 18)) (param "g" (type 19))))
-  (type (;21;) (tuple (type 12) (type 14) (type 15) (type 16) (type 17) (type 18) (type 19)))
-  (type (;22;) (func (result (type 21))))
-  (type (;23;) (variant (case "a" s32) (case "b" float32)))
-  (type (;24;) (variant (case "a" float64) (case "b" float32)))
-  (type (;25;) (variant (case "a" float64) (case "b" u64)))
-  (type (;26;) (variant (case "a" u32) (case "b" s64)))
-  (type (;27;) (variant (case "a" float32) (case "b" s64)))
-  (type (;28;) (tuple float32 u32))
-  (type (;29;) (tuple u32 u32))
-  (type (;30;) (variant (case "a" (type 28)) (case "b" (type 29))))
-  (type (;31;) (tuple (type 23) (type 24) (type 25) (type 26) (type 27) (type 30)))
-  (type (;32;) (func (param "a" (type 23)) (param "b" (type 24)) (param "c" (type 25)) (param "d" (type 26)) (param "e" (type 27)) (param "f" (type 30)) (result (type 31))))
-  (type (;33;) (expected unit unit))
-  (type (;34;) (expected unit (type 0)))
-  (type (;35;) (expected (type 0) unit))
-  (type (;36;) (expected (type 13) (type 13)))
-  (type (;37;) (expected u32 (type 7)))
-  (type (;38;) (list u8))
-  (type (;39;) (expected string (type 38)))
-  (type (;40;) (func (param "a" (type 33)) (param "b" (type 34)) (param "c" (type 35)) (param "d" (type 36)) (param "e" (type 37)) (param "f" (type 39))))
-  (type (;41;) (tuple (type 33) (type 34) (type 35) (type 36) (type 37) (type 39)))
-  (type (;42;) (func (result (type 41))))
-  (type (;43;) (enum "bad1" "bad2"))
-  (type (;44;) (expected s32 (type 43)))
-  (type (;45;) (func (result (type 44))))
-  (type (;46;) (expected unit (type 43)))
-  (type (;47;) (func (result (type 46))))
-  (type (;48;) (expected (type 43) (type 43)))
-  (type (;49;) (func (result (type 48))))
-  (type (;50;) (tuple s32 u32))
-  (type (;51;) (expected (type 50) (type 43)))
-  (type (;52;) (func (result (type 51))))
-  (type (;53;) (option s32))
-  (type (;54;) (func (result (type 53))))
-  (type (;55;) (option (type 43)))
-  (type (;56;) (func (result (type 55))))
-  (type (;57;) (expected u32 s32))
-  (type (;58;) (func (result (type 57))))
+  (type (;19;) (option bool))
```

Comment:
I think this comes from not updating the corresponding `PartialEq` implementation in `wit-component/src/encodinr.rs`mfor `TypeDefKey` with the new `Option` and `Expected` variants.

Can we make that fix so that this file's changes are reverted (one hopes)?

Reply:
Indeed with that fix there are no changes to this file!